library/src/test/java/io/voucherify/android/client/module/ValidationTest.java

```diff
@@ -0,0 +1,124 @@
+package io.voucherify.android.client.module;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.reactivex.schedulers.Schedulers;
+import io.voucherify.android.client.api.VoucherifyApi;
+import io.voucherify.android.client.model.OrderItem;
+import io.voucherify.android.client.model.VoucherResponse;
+import okhttp3.Request;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ValidationTest {
```

Comment:
hmm... I think we invoke it only once per test method in:
https://github.com/voucherifyio/voucherify-android-sdk/pull/37/files#diff-f5efec0e343467b500b6bba0ed31b4e4R46

in this code we just verify if it has been called with given `query`. https://github.com/voucherifyio/voucherify-android-sdk/pull/37/files#diff-f5efec0e343467b500b6bba0ed31b4e4R51

You r right that I do not reset mock after each test method, but in this case it doesn't make any difference.

Test will not pass for sure... it is up to the parameters passed in `queryParams`. If you change them to not match the expected ones it will fail.